src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_SIZE = 100;
+    private Random randomFigure = new Random();
+    private Random randomParameter = new Random();
+    private int dieNumber;
+    private ColorSupplier supplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        dieNumber = randomFigure.nextInt(5) + 1;
+        switch (dieNumber) {
+            case 1:
+                Circle circle = new Circle(randomParameter.nextInt(MAX_RANDOM_SIZE),
+                        supplier.getRandomColor());
+                circle.getArea();
+                return circle;
+            case 2:
+                RightTriangle rightTriangle = new RightTriangle(randomParameter
+                        .nextInt(MAX_RANDOM_SIZE),
+                        randomParameter.nextInt(MAX_RANDOM_SIZE), supplier.getRandomColor());
+                rightTriangle.getArea();
+                return rightTriangle;

```

Comment:
It's better to move each figure creation and configuration to a separate method, like getRandomTriangle() and so on